### PR TITLE
Issue 162

### PR DESCRIPTION
**Pull request by fsanaulla**

### CODE REVIEW

File: src/main/scala/com/hamstoo/daos/MongoContentDao.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import com.hamstoo.utils._
+import org.joda.time.DateTime
+import play.api.Logger
+import reactivemongo.api.BSONSerializationPack.{Reader, Writer}
+import reactivemongo.api.collections.bson.BSONCollection
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  *
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/daos/MongoContentDao.scala]: What is a "content"?  An annotation?  ScalaDoc needs to be enhanced here, but perhaps also it'd be better to call this the MongoAnnotationDao.

File: src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala [hunk 2]
```diff
@@ -18,34 +18,35 @@ import scala.concurrent.Future
...
   case class WeeHighlight(usrId: UUID, id: String, timeFrom: Long, url: String)
-  private val marksColl: Future[BSONCollection] = db map (_ collection "entries")
+
   for {
     c <- futColl
     mc <- marksColl
 
     oldIdx = s"bin-$USR-1-uPref-1"
-    _ = logger.info(s"Dropping index $oldIdx")
+    _ = log.info(s"Dropping index $oldIdx")
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala]: All of our code uses `logger.`  not `log.`.

File: src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala [hunk 3]
```diff
@@ -64,43 +65,23 @@ class MongoHighlightDao(db: Future[DefaultDB]) {
...
+  def update(usr: UUID,
+             id: String,
+             pos: HLPosition,
+             prv: Highlight.Preview,
+             coord: Option[PageCoord]): Future[Highlight] = for {
     c <- futColl
     now = DateTime.now.getMillis
     sel = d :~ USR -> usr :~ ID -> id :~ curnt
-    wr <- c.findAndUpdate(sel, d :~ "$set" -> (d :~ TIMETHRU -> now), fetchNewObject = true)
+    wr <- c findAndUpdate(sel, d :~ "$set" -> (d :~ TIMETHRU -> now), fetchNewObject = true)
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala]: Please don't use inline function notation.  It's okay for DSLs but non-standard for everyday Scala.  In particular, please don't make style changes like this unless they're particularly egregious.

File: src/main/scala/com/hamstoo/daos/MongoInlineNoteDao.scala [hunk 4]
```diff
@@ -2,7 +2,7 @@ package com.hamstoo.daos
 
 import java.util.UUID
 
-import com.hamstoo.models.{InlineNote, Mark, PageCoord}
+import com.hamstoo.models.{InlineNote, InlineNotePosition, Mark, PageCoord}
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/daos/MongoInlineNoteDao.scala]: `InlineNotePosition` should now be `InlineNote.Position` though perhaps those changes just aren't merged yet.  Same goes for `HLPosition` -> `Highlight.Position`.

File: src/main/scala/com/hamstoo/daos/auth/MongoAuthDao.scala [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import play.api.Logger
+import reactivemongo.api.collections.bson.BSONCollection
+import reactivemongo.bson.BSONDocument
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe._
+
+abstract class MongoAuthDao[A <: AuthInfo: ClassTag: TypeTag](coll: Future[BSONCollection], log: Logger) extends DelegableAuthInfoDAO[A] {
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/daos/auth/MongoAuthDao.scala]: ScalaDoc missing.

File: src/main/scala/com/hamstoo/models/Content.scala [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package com.hamstoo.models
+
+import java.util.UUID
+
+import com.github.dwickern.macros.NameOf.nameOf
+
+trait Content {
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/models/Content.scala]: Again, what is a "content"?  Should it be an "annotation"?

File: src/test/scala/com/hamstoo/daos/MongoInlineNoteDaoTests.scala [hunk 7]
```diff
@@ -2,42 +2,46 @@ package com.hamstoo.daos
...
-    val markId = generateDbId(Mark.ID_LENGTH)
+  // because of dropping "bin-usrId-1-uPref-1" index
+  it should "* (UNIT) retrieve inline note by id" ignore {
+    notesDao.retrieve(c.usrId, c.id).futureValue.value shouldEqual c
+  }
 
-    val c1 = InlineNote(usrId, markId = markId, pos = InlineNote.Position("sdassd", "sdassd", 0, 0), pageCoord = Some(PageCoord(0.5, 0.5)))
-    val c2 = InlineNote(usrId, markId = markId, pos = InlineNote.Position("sdassd", "sdassd", 0, 0), pageCoord = Some(PageCoord(0.6, 0.5)))
-    val c3 = InlineNote(usrId, markId = markId, pos = InlineNote.Position("sdassd", "sdassd", 0, 0), pageCoord = Some(PageCoord(0.4, 0.8)))
+  it should "* (UNIT) retrieve inline note by markId" in {
```
Comment (fcrimins) [inline src/test/scala/com/hamstoo/daos/MongoInlineNoteDaoTests.scala]: No need to use the `*` at the beginning of "should" strings anymore.  Those were only there for use with Specs2 which had poor formatting when running tests from IntelliJ.

File: src/main/scala/com/hamstoo/models/Position.scala [hunk 8]
```diff
@@ -0,0 +1,22 @@
+package com.hamstoo.models
+
+sealed trait Position
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/models/Position.scala]: Okay to have this sealed trait--nice in fact--just move the extended case classes back to where they were before, inside Highlight and InlineNote companion objects.

File: src/main/scala/com/hamstoo/services/HighlightsIntersectionService.scala [hunk 9]
```diff
@@ -54,23 +54,23 @@ class HighlightsIntersectionService(hlightsDao: MongoHighlightDao)(implicit ec:
             nHl.copy(pos = pos, preview = prv, pageCoord = coord)
           }
         }
-        hlightsDao create h map (_ => h)
+        hlightsDao insert h map (_ => h)
```
Comment (fcrimins) [inline src/main/scala/com/hamstoo/services/HighlightsIntersectionService.scala:57]: Good change of function name here.  `insert` is more standard.

File: src/test/scala/com/hamstoo/daos/auth/MongoAuth1InfoDaoTest.scala [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+import com.hamstoo.daos.MongoUserDao
+import com.hamstoo.models.{Profile, User, UserData}
+import com.hamstoo.test.env.MongoEnvironment
+import com.hamstoo.test.{FlatSpecWithMatchers, FutureHandler}
+import com.hamstoo.utils.TestHelper
+import com.mohiva.play.silhouette.api.LoginInfo
+import com.mohiva.play.silhouette.impl.providers.OAuth1Info
+import org.scalatest.OptionValues
+
+class MongoAuth1InfoDaoTest
```
Comment (fcrimins) [inline src/test/scala/com/hamstoo/daos/auth/MongoAuth1InfoDaoTest.scala:14]: Love all the tests!

File: src/test/scala/com/hamstoo/utils/TestHelper.scala [hunk 11]
```diff
@@ -6,7 +6,6 @@ import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 import scala.util.{Failure, Success}
 
-// https://github.com/etorreborre/specs2/blob/SPECS2-3.8.9/examples/src/test/scala/examples/UnitSpec.scala
 trait TestHelper extends DataInfo {
```
Comment (fcrimins) [inline src/test/scala/com/hamstoo/utils/TestHelper.scala]: ScalaDoc missing.

File: testkit/src/main/scala/com/hamstoo/test/env/MongoEnvironment.scala [hunk 12]
```diff
@@ -1,14 +1,11 @@
...
 
 import com.github.simplyscala.{MongoEmbedDatabase, MongodProps}
 import de.flapdoodle.embed.mongo.distribution.Version
 import org.scalatest.{BeforeAndAfterAll, Suite}
-import play.api.Logger
 
-/**
-  * Trait that start Fake MongoDB instance
-  */
 trait MongoEnvironment extends MongoEmbedDatabase with BeforeAndAfterAll {
```
Comment (fcrimins) [inline testkit/src/main/scala/com/hamstoo/test/env/MongoEnvironment.scala]: ScalaDoc missing.

**Review by fcrimins — APPROVED** [review]
Changes look really nice, just a few small things to address.

File: src/main/scala/com/hamstoo/daos/MongoContentDao.scala (hunk 1 above)
Comment (fsanaulla) [reply src/main/scala/com/hamstoo/daos/MongoContentDao.scala]: Renamed to MongoAnnotationDao, scaladocs added

File: src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala (hunk 2 above)
Comment (fsanaulla) [reply src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala]: Done

File: src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala (hunk 3 above)
Comment (fsanaulla) [reply src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala]: Done

File: src/main/scala/com/hamstoo/daos/MongoInlineNoteDao.scala (hunk 4 above)
Comment (fsanaulla) [reply src/main/scala/com/hamstoo/daos/MongoInlineNoteDao.scala]: Done.

File: src/main/scala/com/hamstoo/daos/auth/MongoAuthDao.scala (hunk 5 above)
Comment (fsanaulla) [reply src/main/scala/com/hamstoo/daos/auth/MongoAuthDao.scala]: Added

File: src/main/scala/com/hamstoo/models/Content.scala (hunk 6 above)
Comment (fsanaulla) [reply src/main/scala/com/hamstoo/models/Content.scala]: moved to Annotation trait

File: src/test/scala/com/hamstoo/utils/TestHelper.scala (hunk 11 above)
Comment (fsanaulla) [reply src/test/scala/com/hamstoo/utils/TestHelper.scala]: Added

File: testkit/src/main/scala/com/hamstoo/test/env/MongoEnvironment.scala (hunk 12 above)
Comment (fsanaulla) [reply testkit/src/main/scala/com/hamstoo/test/env/MongoEnvironment.scala]: Added

File: src/main/scala/com/hamstoo/models/Position.scala (hunk 8 above)
Comment (fsanaulla) [reply src/main/scala/com/hamstoo/models/Position.scala]: Moved.

File: src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala (hunk 3 above)
Comment (crimzie) [reply src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala]: What do you call "inline notation"? Maybe you meant infix notation?

Comment (fcrimins) [reply src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala]: Yes, I meant infix notation, not inline notation.

Comment (crimzie) [reply src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala]: So is the current policy to not use infix notation wherever possible? Asking just for me to be aware

Comment (fcrimins) [reply src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala]: Yes, let's say this is the current policy but (1) please do not rewrite any code to achieve this purpose and (2) I think it's fine to use in some standard places, like when performing a functional map.  For example, both of these are allowed `x map { ... }` and `x.map(...)` but this is not `logger info "msg"`.  Most importantly, long sequences of identifiers should always be delineated with dots and parens.

Comment (crimzie) [reply src/main/scala/com/hamstoo/daos/MongoHighlightDao.scala]: Ok

**Pull request merged**